lib/src/buttons/help_button.dart

```diff
@@ -0,0 +1,208 @@
+  final double? pressedOpacity;
+
+  /// The alignment of the button's [child].
+  ///
+  /// Typically buttons are sized to be just big enough to contain the child and its
+  /// [padding]. If the button's size is constrained to a fixed size, for example by
+  /// enclosing it with a [SizedBox], this property defines how the child is aligned
+  /// within the available space.
+  ///
+  /// Always defaults to [Alignment.center].
+  final AlignmentGeometry alignment;
+
+  /// Whether the button is enabled or disabled. Buttons are disabled by default. To
+  /// enable a button, set its [onPressed] property to a non-null value.
+  bool get enabled => onPressed != null;
+
+  @override
+  _HelpButtonState createState() => _HelpButtonState();
+}
+
+class _HelpButtonState extends State<HelpButton>
+    with SingleTickerProviderStateMixin {
+  // Eyeballed values. Feel free to tweak.
+  static const Duration kFadeOutDuration = Duration(milliseconds: 10);
+  static const Duration kFadeInDuration = Duration(milliseconds: 100);
+  final Tween<double> _opacityTween = Tween<double>(begin: 1.0);
+
+  late AnimationController _animationController;
+  late Animation<double> _opacityAnimation;
+
+  @override
+  void initState() {
+    super.initState();
+    _animationController = AnimationController(
+      duration: const Duration(milliseconds: 200),
+      value: 0.0,
+      vsync: this,
+    );
+    _opacityAnimation = _animationController
+        .drive(CurveTween(curve: Curves.decelerate))
+        .drive(_opacityTween);
+    _setTween();
+  }
+
+  @override
+  void didUpdateWidget(HelpButton old) {
+    super.didUpdateWidget(old);
+    _setTween();
+  }
+
+  void _setTween() {
+    _opacityTween.end = widget.pressedOpacity ?? 1.0;
+  }
+
+  @override
+  void dispose() {
+    _animationController.dispose();
+    super.dispose();
+  }
+
+  bool _buttonHeldDown = false;
+
+  void _handleTapDown(TapDownDetails event) {
+    if (!_buttonHeldDown) {
+      _buttonHeldDown = true;
+      _animate();
+    }
+  }
+
+  void _handleTapUp(TapUpDetails event) {
+    if (_buttonHeldDown) {
+      _buttonHeldDown = false;
+      _animate();
+    }
+  }
+
+  void _handleTapCancel() {
+    if (_buttonHeldDown) {
+      _buttonHeldDown = false;
+      _animate();
+    }
+  }
+
+  void _animate() {
+    if (_animationController.isAnimating) return;
+    final bool wasHeldDown = _buttonHeldDown;
+    final TickerFuture ticker = _buttonHeldDown
+        ? _animationController.animateTo(1.0, duration: kFadeOutDuration)
+        : _animationController.animateTo(0.0, duration: kFadeInDuration);
+    ticker.then<void>((void value) {
+      if (mounted && wasHeldDown != _buttonHeldDown) _animate();
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final bool enabled = widget.enabled;
+    final MacosThemeData theme = MacosTheme.of(context);
+    final Color? backgroundColor = widget.color == null
+        ? theme.helpButtonTheme.color
```

Comment:
This color should also be handled inside of a `DynamicColorX.macosResolve`:

```dart
DyanmicColorX.macosResolve(widget.color ?? theme.helpButtonTheme.color)
```

Reply:
Can you elaborate on this a bit more. I mean is this the way you want it ? 

final Color? backgroundColor = widget.color == null
        ? DyanmicColorX.macosResolve(widget.color ?? theme.helpButtonTheme.color)
        : CupertinoDynamicColor.maybeResolve(widget.color, context);

I am getting error when I am using this way. Error is "DynamicColorX is undefined class".

Reply:
It's not a class but an extension. https://github.com/GroovinChip/macos_ui/blob/dev/lib/src/styles/colors.dart

```dart
final Color? backgroundColor = DynamicColorX.macosResolve(widget.color ?? theme.helpButtonTheme.color, context)
```